com.unity.perception/Runtime/GroundTruth/SimulationState.cs

```diff
@@ -615,10 +622,61 @@ public AsyncAnnotation ReportAnnotationAsync(AnnotationDefinition annotationDefi
             ReportAsyncAnnotationResult<T>(asyncAnnotation, filename, jArray);
         }
 
-        public void ReportAsyncAnnotationResult<T>(AsyncAnnotation asyncAnnotation, string filename = null, IEnumerable<T> values = null)
+        [SuppressMessage("ReSharper", "PossibleInvalidOperationException")]
+        public class QuaternionConverter : JsonConverter<Quaternion>
         {
-            var jArray = values == null ? null : JArray.FromObject(values);
+            public override void WriteJson(JsonWriter writer, Quaternion quaternion, JsonSerializer serializer)
```

Comment:
That's an interesting approach. We have some existing code for these types which is pretty similar in DatasetJsonUtility. Give it a quick look to see if it's worth having both versions.